src/VirtoCommerce.ExperienceApiModule.DigitalCatalog/Schemas/CatalogDiscountType.cs

```diff
@@ -16,7 +16,7 @@ public CatalogDiscountType(IMediator mediator, IDataLoaderContextAccessor dataLo
             {
                 var loader = dataLoader.Context.GetOrAddBatchLoader<string, Promotion>("promotionsLoader", (ids) => LoadPromotionsAsync(mediator, ids));
 
-                return await loader.LoadAsync(context.Source.PromotionId);
+                return await loader.LoadAsync(context.Source.PromotionId).GetResultAsync();
```

Comment:
Can you split this line into two to prevent possible deadlocks?

Reply:
done